### ISSUE TITLE
Suppress key_data in debug output

### ISSUE THREAD

**Issue by jtimberman** [chef/chef-provisioning#171]
The debug output shows the RSA private key:

```
[2014-09-12T08:53:54-06:00] DEBUG: Opening SSH connection to ubuntu@10.10.10.111 with options {:timeout=>10, :auth_methods=>["publickey"], :keys_only=>true, :host_key_alias=>"i-db46f7d4.AWS", :key_data=>["-----BEGIN RSA PRIVATE KEY----- private key content is here-----END RSA PRIVATE KEY-----\n"]}
```

It should be suppressed in the output, so noone accidentally pastes this in a public or semi-public forum (pastie, gist, etc) when getting help/troubleshooting.

**Comment by tyler-ball** [comment]
Closed by #245

**Comment by jtimberman** [comment]
:cake:
